packages/dds/sequence/src/test/revertibles.spec.ts

```diff
@@ -508,3 +489,322 @@ describe("Sequence.Revertibles with Remote Edits", () => {
+			collection2 = sharedString2.getIntervalCollection("test");
+		});
+
+		it("handles remote remove of same range", () => {
+			sharedString.insertText(0, "hello world");
+
+			sharedString.on("sequenceDelta", (op) => {
+				if (op.isLocal) {
+					appendSharedStringDeltaToRevertibles(sharedString, op, revertibles);
+				}
+			});
+
+			collection.add(2, 4, IntervalType.SlideOnRemove);
+			containerRuntimeFactory.processAllMessages();
+
+			sharedString2.removeRange(0, 6);
+			sharedString.removeRange(0, 6);
+			assert.equal(sharedString.getText(), "world");
+			assert.equal(sharedString2.getText(), "world");
+			containerRuntimeFactory.processAllMessages();
+
+			assert.equal(revertibles.length, 1, "revertibles.length is not 1");
+			revertSharedStringRevertibles(sharedString, revertibles.splice(0));
+
+			assert.equal(sharedString.getText(), "hello world");
+			assertIntervals(sharedString, collection, [{ start: 2, end: 4 }]);
+			containerRuntimeFactory.processAllMessages();
+			assertIntervals(sharedString2, collection2, [{ start: 2, end: 4 }]);
+		});
+		it("ignores remote interval move", () => {
+			sharedString.insertText(0, "hello world");
+
+			sharedString.on("sequenceDelta", (op) => {
+				if (op.isLocal) {
+					appendSharedStringDeltaToRevertibles(sharedString, op, revertibles);
+				}
+			});
+
+			const interval = collection.add(2, 4, IntervalType.SlideOnRemove);
+			containerRuntimeFactory.processAllMessages();
+
+			sharedString.removeRange(0, 6);
+			sharedString2.getIntervalCollection("test").change(interval.getIntervalId(), 3, 8);
+			containerRuntimeFactory.processAllMessages();
+
+			assert.equal(revertibles.length, 1, "revertibles.length is not 1");
+			revertSharedStringRevertibles(sharedString, revertibles.splice(0));
+
+			assert.equal(sharedString.getText(), "hello world");
+			// start moved within deleted range is restored, end moved outside is not
+			assertIntervals(sharedString, collection, [{ start: 2, end: 4 }]);
+			containerRuntimeFactory.processAllMessages();
+			assertIntervals(sharedString2, collection2, [{ start: 2, end: 4 }]);
+		});
+		it("handles remote interval delete", () => {
+			sharedString.insertText(0, "hello world");
+
+			sharedString.on("sequenceDelta", (op) => {
+				if (op.isLocal) {
+					appendSharedStringDeltaToRevertibles(sharedString, op, revertibles);
+				}
+			});
+
+			const interval = collection.add(2, 4, IntervalType.SlideOnRemove);
+			containerRuntimeFactory.processAllMessages();
+
+			sharedString.removeRange(0, 6);
+			sharedString2
+				.getIntervalCollection("test")
+				.removeIntervalById(interval.getIntervalId());
+			containerRuntimeFactory.processAllMessages();
+
+			assert.equal(revertibles.length, 1, "revertibles.length is not 1");
+			revertSharedStringRevertibles(sharedString, revertibles.splice(0));
+
+			assert.equal(sharedString.getText(), "hello world");
+			assert.equal(collection.getIntervalById(interval.getIntervalId()), undefined);
```

Comment:
I think you can use assertIntervals to check this case as well, the last parameter would just need to be an empty array.